lib/Doctrine/MongoDB/Aggregation/Expr.php

```diff
@@ -950,6 +1190,54 @@ private function requiresCurrentField($method = null)
     }
 
     /**
+     * @param string $method
+     *
+     * @throws \BadMethodCallException if there is no current switch operator
+     */
+    private function requiresSwitchStatement($method = null)
+    {
+        $message = ($method ?: 'This method') . ' requires a valid switch statement (call switch() first).';
+
+        if ($this->currentField) {
+            if (!isset($this->expr[$this->currentField]['$switch'])) {
```

Comment:
This might be me leaking coding standards from other projects into here - I'd have to check. If you ask me I'd do away with the space 😛